ICE:  'expected `NodeId` to be lowered already for res Local

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.

http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/

-->

### Code

```Rust
fn main () {
  match x {
    Some :: <v<>>(v)  => ()
  }
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.52.0-nightly (51748a8fc 2021-03-05)
binary: rustc
commit-hash: 51748a8fc77283914d4135f31b5966a407208187
commit-date: 2021-03-05
host: x86_64-unknown-linux-gnu
release: 1.52.0-nightly
LLVM version: 12.0.0
```

### Error output

```
error[E0425]: cannot find value `x` in this scope
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:2:9
  |
2 |   match x {
  |         ^ not found in this scope

error[E0573]: expected type, found local variable `v`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:14
  |
3 |     Some :: <v<>>(v)  => ()
  |              ^^^ not a type

thread 'rustc' panicked at 'expected `NodeId` to be lowered already for res Local(
    NodeId(20),
)', compiler/rustc_ast_lowering/src/lib.rs:715:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.52.0-nightly (51748a8fc 2021-03-05) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to 2 previous errors

Some errors have detailed explanations: E0425, E0573.
For more information about an error, try `rustc --explain E0425`.
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
error[E0425]: cannot find value `x` in this scope
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:2:9
  |
2 |   match x {
  |         ^ not found in this scope

error[E0573]: expected type, found local variable `v`
 --> perses_node_priority_with_dfs_delta_reduced_mutant.rs:3:14
  |
3 |     Some :: <v<>>(v)  => ()
  |              ^^^ not a type

thread 'rustc' panicked at 'expected `NodeId` to be lowered already for res Local(
    NodeId(20),
)', compiler/rustc_ast_lowering/src/lib.rs:715:17
stack backtrace:
   0: rust_begin_unwind
             at /rustc/51748a8fc77283914d4135f31b5966a407208187/library/std/src/panicking.rs:493:5
   1: std::panicking::begin_panic_fmt
             at /rustc/51748a8fc77283914d4135f31b5966a407208187/library/std/src/panicking.rs:435:5
   2: rustc_ast_lowering::LoweringContext::lower_node_id_generic
   3: rustc_ast_lowering::path::<impl rustc_ast_lowering::LoweringContext>::lower_path_segment
   4: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
   5: rustc_ast_lowering::path::<impl rustc_ast_lowering::LoweringContext>::lower_qpath
   6: rustc_ast_lowering::LoweringContext::lower_ty_direct
   7: rustc_ast_lowering::LoweringContext::lower_generic_arg
   8: <smallvec::SmallVec<A> as core::iter::traits::collect::Extend<<A as smallvec::Array>::Item>>::extend
   9: rustc_ast_lowering::path::<impl rustc_ast_lowering::LoweringContext>::lower_angle_bracketed_parameter_data
  10: rustc_ast_lowering::path::<impl rustc_ast_lowering::LoweringContext>::lower_path_segment
  11: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
  12: rustc_ast_lowering::path::<impl rustc_ast_lowering::LoweringContext>::lower_qpath
  13: rustc_ast_lowering::pat::<impl rustc_ast_lowering::LoweringContext>::lower_pat::{{closure}}
  14: rustc_ast_lowering::expr::<impl rustc_ast_lowering::LoweringContext>::lower_arm
  15: rustc_ast_lowering::Arena::alloc_from_iter
  16: rustc_ast_lowering::expr::<impl rustc_ast_lowering::LoweringContext>::lower_expr_mut::{{closure}}
  17: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
  18: <core::iter::adapters::flatten::Flatten<I> as core::iter::traits::iterator::Iterator>::next
  19: <smallvec::SmallVec<A> as core::iter::traits::collect::Extend<<A as smallvec::Array>::Item>>::extend
  20: rustc_arena::cold_path
  21: rustc_ast_lowering::item::<impl rustc_ast_lowering::LoweringContext>::lower_fn_body_block
  22: rustc_ast_lowering::item::<impl rustc_ast_lowering::LoweringContext>::lower_item
  23: rustc_ast_lowering::LoweringContext::with_hir_id_owner
  24: rustc_ast_lowering::lower_crate
  25: rustc_interface::passes::BoxedResolver::access::{{closure}}
  26: rustc_interface::passes::configure_and_expand::{{closure}}
  27: rustc_interface::passes::BoxedResolver::access
  28: rustc_interface::queries::Queries::lower_to_hir
  29: rustc_interface::queries::Queries::global_ctxt
  30: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter
  31: rustc_span::with_source_map
  32: rustc_interface::interface::create_compiler_and_run
  33: rustc_span::with_session_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.52.0-nightly (51748a8fc 2021-03-05) running on x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
error: aborting due to 2 previous errors

Some errors have detailed explanations: E0425, E0573.
For more information about an error, try `rustc --explain E0425`.
```

</p>
</details>

--- a/src/test/ui/match/issue-82866.rs
+++ b/src/test/ui/match/issue-82866.rs
@@ -1,0 +1,7 @@
+fn main() {
+    match x {
+        //~^ ERROR cannot find value `x` in this scope
+        Some::<v>(v) => (),
+        //~^ ERROR cannot find type `v` in this scope
+    }
+}

--- a/src/test/ui/match/issue-82866.stderr
+++ b/src/test/ui/match/issue-82866.stderr
@@ -1,0 +1,16 @@
+error[E0425]: cannot find value `x` in this scope
+  --> $DIR/issue-82866.rs:2:11
+   |
+LL |     match x {
+   |           ^ not found in this scope
+
+error[E0412]: cannot find type `v` in this scope
+  --> $DIR/issue-82866.rs:4:16
+   |
+LL |         Some::<v>(v) => (),
+   |                ^ not found in this scope
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0412, E0425.
+For more information about an error, try `rustc --explain E0412`.

